app/feature/feature-addon-purchase/src/main/kotlin/com/hedvig/android/feature/addon/purchase/data/GetTravelAddonOfferUseCase.kt

```diff
@@ -21,13 +30,78 @@ internal class GetTravelAddonOfferUseCaseImpl(
   private val featureManager: FeatureManager,
 ) : GetTravelAddonOfferUseCase {
   override suspend fun invoke(id: String): Either<ErrorMessage, TravelAddonOffer> {
-    // todo: REMOVE MOCK!
     return either {
-      mockWithUpgrade
+      val isAddonFlagOn = featureManager.isFeatureEnabled(Feature.TRAVEL_ADDON).first()
+      if (!isAddonFlagOn) {
+        logcat(LogPriority.ERROR) { "Tried to start UpsellAddonOfferMutation but addon feature flag is off" }
+        raise(ErrorMessage())
+      } else {
```

Comment:
Can remove this `else` completely since `raise` will short-circuit this function and exit early in that case.
That way everything below here will be indented 2 spaces less